lib/config-generator.js

```diff
@@ -0,0 +1,261 @@
+                choices: [
+                    { message: "React", name: "react" },
+                    { message: "Vue.js", name: "vue" },
+                    { message: "None of these", name: "none" }
+                ]
+            },
+            {
+                type: "select",
+                name: "language",
+                message: "Does your project use TypeScript?",
+                choices: [
+                    { message: "No", name: "javascript" },
+                    { message: "Yes", name: "typescript" }
+                ],
+                initial: 0
+            },
+            {
+                type: "multiselect",
+                name: "env",
+                message: "Where does your code run?",
+                hint: "(Press <space> to select, <a> to toggle all, <i> to invert selection)",
+                initial: 0,
+                choices: [
+                    { message: "Browser", name: "browser" },
+                    { message: "Node", name: "node" }
+                ]
+            }
+        ];
+
+        const answers = await enquirer.prompt(questions);
+
+        Object.assign(this.answers, answers);
+
+        if (answers.purpose === "style") {
+            const choices = this.answers.language === "javascript" ? jsStyleGuides : tsStyleGuides;
+            const styleguideAnswer = await enquirer.prompt({
+                type: "select",
+                name: "styleguide",
+                message: "Which style guide do you want to follow?",
+                choices,
+                result: choice => choices.find(it => it.name === choice).value
+            });
+
+            Object.assign(this.answers, styleguideAnswer);
+        }
+    }
+
+    /**
+     * Calculate the configuration based on the user's answers.
+     * @returns {void}
+     */
+    calc() {
+        const isESMModule = isPackageTypeModule(this.packageJsonPath);
+
+        this.result.configFilename = isESMModule ? "eslint.config.js" : "eslint.config.mjs";
+
+        let importContent = "";
+        const helperContent = `import path from "path";
+import { fileURLToPath } from "url";
+import { FlatCompat } from "@eslint/eslintrc";
+import pluginJs from "@eslint/js";
+
+// mimic CommonJS variables -- not needed if using CommonJS
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const compat = new FlatCompat({baseDirectory: __dirname, recommendedConfig: pluginJs.configs.recommended});
+`;
+        let exportContent = "";
+        let languageOptionsContent = "";
+        let needCompatHelper = false;
+
+        if (this.answers.moduleType === "commonjs" || this.answers.moduleType === "script") {
+            exportContent += `  {files: ["**/*.js"], languageOptions: {sourceType: "${this.answers.moduleType}"}},\n`;
+        }
+
+        if (this.answers.purpose === "syntax") {
+
+            // no need to install any plugin
+        } else if (this.answers.purpose === "problem") {
```

Comment:
```suggestion
        } else if (this.answers.purpose === "problems") {
```

To match the `name` in the definition above:

```js
                    { message: "To check syntax and find problems", name: "problems" },
```